server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeService.java

```diff
@@ -183,6 +183,17 @@ public RepositoriesMetadata updateRepositoriesMetadata(DiscoveryNode joiningNode
                 boolean repositoryAlreadyPresent = false;
                 for (RepositoryMetadata existingRepositoryMetadata : existingRepositories.repositories()) {
                     if (newRepositoryMetadata.name().equals(existingRepositoryMetadata.name())) {
+                        // This is to handle cases where-in the during a previous node-join attempt if the publish operation succeeded but
+                        // the commit operation failed, the cluster-state may have the repository metadata which is not applied into the
+                        // repository service. This may lead to assertion failures down the line.
+                        if (!repositoriesService.get().isRepositoryPresent(newRepositoryMetadata.name())) {
```

Comment:
Should we validate the the repositories using the metadata in cluster state instead of fetching from repositories service ?

Reply:
I think the following method `ensureValidSystemRepositoryUpdate` fetches the repository object corresponding to the repo name and validates the `RestrictedSystemRepositorySettings` corresponding to the same. I believe we would need the repositories from the RepositoryService to validate.